src/mongo/client/dbclient_rs.cpp

```diff
@@ -426,10 +427,25 @@ namespace {
                     break;
                 }
 
-                conn->auth( params );
+                BSONObj interposedParams;
```

Comment:
Maybe start with BSONObj interposedParams = params. That way, you can eliminate the else block below.